### ISSUE TITLE
spineItem object with "dynamically" attached paginationInfo data in reflowable view (not FXL or scroll)

### ISSUE THREAD

**Issue by danielweck** [readium/readium-shared-js#280]
`loadSpineItem()` in reflowable_view:
https://github.com/readium/readium-shared-js/blob/develop/js/views/reflowable_view.js#L204

```javascript
// TODO: this is a dirty hack!!
_currentSpineItem.paginationInfo = _paginationInfo;             
```

The `paginationInfo` field is missing in the one_page_view (which drives both FXL and scroll modes!) `loadSpineItem()`:
https://github.com/readium/readium-shared-js/blob/develop/js/views/one_page_view.js#L774

This is needed in the highlight plugin:
https://github.com/readium/readium-shared-js/blob/develop/plugins/highlights/controller.js#L58
https://github.com/readium/readium-shared-js/blob/develop/plugins/highlights/controller.js#L213
https://github.com/readium/readium-shared-js/blob/develop/plugins/highlights/controller.js#L569
https://github.com/readium/readium-shared-js/blob/develop/plugins/highlights/controller.js#L576

```javascript
var isVerticalWritingMode = this.context.paginationInfo().isVerticalWritingMode;
```

Note that cfi_navigation_logic gets the `paginationInfo` via its `options` parameter:
https://github.com/readium/readium-shared-js/blob/develop/js/views/cfi_navigation_logic.js#L40

**Comment by danielweck** [comment]
Note that `getPaginationInfo()` is actually exposed in several API points:

reader_view:
https://github.com/readium/readium-shared-js/blob/develop/js/views/reader_view.js#L1545

fixed_view:
https://github.com/readium/readium-shared-js/blob/develop/js/views/fixed_view.js#L568

scroll_view:
https://github.com/readium/readium-shared-js/blob/develop/js/views/scroll_view.js#L1116

reflowable_view:
https://github.com/readium/readium-shared-js/blob/develop/js/views/reflowable_view.js#L779

...but in the case of reflowable_view, `_paginationInfo` is different than `getPaginationInfo()`!
https://github.com/readium/readium-shared-js/blob/develop/js/views/reflowable_view.js#L76

==> needs clarifying.

**Comment by bradleygore** [comment]
I believe that the fixed view has a same/similar issue, so posting this here instead of a new issue. If it needs a new issue, please advise and I'll be happy to do so.

I'm using Readium v0.23.0 and have noticed that highlights are broken for the fixed view type as well. What I'm seeing is the same lack of any result from the `getPaginationInfo()` method call.

If you look in reflowable_view:loadSpineItem()
https://github.com/readium/readium-shared-js/blob/develop/js/views/reflowable_view.js#L207
You can see where the `_currentSpineItem.paginationInfo is being set.

However, it's not being set anywhere on the spine item in the fixed view.

I'm looking to work on this issue and get a PR in, but am unsure of the appropriate approach - again, super new to Readium.

Should it be the responsibility of the view to provide paginationInfo (I mean, what if the epub doesn't even need pagination?) or should the highlight plugin be responsible to null-check and fallback to a default or something?

**Comment by danielweck** [comment]
Related bug: https://github.com/readium/readium-shared-js/issues/315 ?

**Comment by bradleygore** [comment]
@danielweck Yep, exactly. Didn't see that open bug already existed when making my comment.

**Comment by danielweck** [comment]
Related (duplicated) issue? https://github.com/readium/readium-shared-js/issues/314

**Comment by entn-at** [comment]
I just ran into the same issue and wanted to ask if anyone has already come up with a solution, or if I should try to take a stab at it myself? In the latter case, I'd be grateful for any pointers/advice (I'm currently stepping through the code using remote debugging in the SDKLauncher-iOS app).
